src/Types/ContextFactory.php

```diff
@@ -70,6 +70,26 @@ public function createForNamespace($namespace, $fileContents)
                 case T_NAMESPACE:
                     $currentNamespace = $this->parseNamespace($tokens);
                     break;
+                case T_CLASS:
+                    // Fast-forward the iterator through the class so that any
+                    // T_USE tokens found within are skipped - these are not
+                    // valid namespace use statements so should be ignored.
+                    $braceLevel = 0;
+                    $firstBraceFound = false;
+                    while ($tokens->valid() && ($braceLevel > 0 || !$firstBraceFound)) {
+                        if ($tokens->current() === '{') {
```

Comment:
@asgrim It's perfectly fine PHP and will work. What I'm saying is that use statements only affect all classes *after* them. So if you have this code:

```php
class A extends X {}
use Foo\X;
class B extends X {}
```

Then class `A` will extend `X`, while class `B` extends `Foo\X`. So if you want to correctly handle this, you have to pass in the class name you're considering (rather than just its namespace name) and only return the aliases that are in effect at the time of the definition of the class.

(So yes, you're right, I'm just pointing out an extra caveat why it won't work out of the box, even if you properly do the skipping here.)